### ISSUE TITLE
Lots of links have extra spaces around them

### ISSUE THREAD

**Issue by ChristopherChudzicki** [mitodl/ocw-hugo-themes#949]
*Re-wrote this to make more clear that multiple things are affected.*

A **lot** of content in OCW has extra whitespace (try viewing the raw HTML). This usually isn't visible to users, but in a few places it is. Here are some examples:

- **Instructors in Featured Course Cards:** on https://ocw.mit.edu

    <img width="200" alt="Screen Shot 2022-11-02 at 10 13 39 AM" src="https://user-images.githubusercontent.com/9010790/199512681-584a6ca9-ebd3-4d87-8e3e-fbbbc96e0f05.png">
- **After "FAQs"on Download page:** https://ocw.mit.edu/courses/res-18-010-a-2020-vision-of-linear-algebra-spring-2020/download/

    <img width="500" alt="Screen Shot 2022-11-07 at 9 58 42 AM" src="https://user-images.githubusercontent.com/9010790/200341917-64074690-3fd9-44b9-9636-a2a67efb1b75.png"> 
- Add more examples...

### Scope
Basically everything using `link.html` partial has this issue, though it's only noticeable if there's a non-whitespace character after or before the link (punctuation or a parenthesis, usually).

Probably an issue with other shortcodes, too.

**Comment by gumaerc** [comment]
@ChristopherChudzicki I think this is reasonable. We should likely have a PR that goes through the whole repo and sets the whitespace stripping delimiters wherever it makes sense to do so.

**Comment by Wassaf-Shahzad** [comment]
@ChristopherChudzicki Should the PR to convent `{{ }}` to `{{- -}}` be a part of this issue or a separate issue ?

In regards to this issue, I have identified that the cause for extra spaces was the delimiter we were adding through hugo rather then adding it using CSS. I have a PR up for this issue resolving issue in `course_card_topics` and `course_card_instrutors`

**Comment by ChristopherChudzicki** [comment]
> Should the PR to convent {{ }} to {{- -}} be a part of this issue or a separate issue ?

@Wassaf-Shahzad I would *expect* yes, because I think that's really the underlying problem.

I'd be happy to look a branch with your CSS fix, but I suspect it's specific to the instructor list, whereas the instructor list issue really just illustrative of a bigger issue.

For example, below there is an extra space between "FAQs" the period ".", and we probably don't want to use CSS in this case.

<img width="847" alt="Screen Shot 2022-11-07 at 9 46 56 AM" src="https://user-images.githubusercontent.com/9010790/200339009-8f695377-3232-43b1-b1de-8d00c29325e1.png">

(That's https://ocw.mit.edu/courses/res-18-010-a-2020-vision-of-linear-algebra-spring-2020/download/).

I suspect we will find a lot of these, and I *believe* the underlying issue is how we are using Go templates. (E.g., we've had the issue before with resource links https://github.com/mitodl/ocw-hugo-themes/pull/878).

**Comment by ChristopherChudzicki** [comment]
> Should the PR to convent {{ }} to {{- -}} be a part of this issue or a separate issue ?

@Wassaf-Shahzad I would *expect* yes, because I think that's really the underlying problem.

I'd be happy to look a branch with your CSS fix, but I suspect it's specific to the instructor list, whereas the instructor list issue really just illustrative of a bigger issue.

For example, below there is an extra space between "FAQs" the period ".", and we probably don't want to use CSS in this case.

<img width="847" alt="Screen Shot 2022-11-07 at 9 46 56 AM" src="https://user-images.githubusercontent.com/9010790/200339009-8f695377-3232-43b1-b1de-8d00c29325e1.png">

(That's https://ocw.mit.edu/courses/res-18-010-a-2020-vision-of-linear-algebra-spring-2020/download/).

I suspect we will find a lot of these, and I *believe* the underlying issue is how we are using Go templates. (E.g., we've had the issue before with resource links https://github.com/mitodl/ocw-hugo-themes/pull/878).

**Comment by ChristopherChudzicki** [comment]
> We should likely have a PR that goes through the whole repo and sets the whitespace stripping delimiters wherever it makes sense to do so.

I would be pretty hesitant to do this in a single PR. It's easy to make very consequential mistakes, e.g., accidentally stripping space between classnames. 

I really did not understand how templates were working with whitespace until I started experimenting with templates on https://repeatit.io/. In trying to understand this better, I made three small demos that I'd like to share:
1. [Only {{ }} is bad](https://repeatit.io/#/share/eyJ0ZW1wbGF0ZSI6Int7LypcbkNocmlzIENodWR6aWNraSwgTm92IDE1LCAyMDIyXG5cbk9ubHkge3sgfX0gPSBMb3RzIG9mIGV4dHJhIHdoaXRlc3BhY2UhXG4oVGhpcyBpcyBjdXJyZW50IE9DVyBIdWdvIFRoZW1lcyBwYXJhZGlnbSlcbiovfX08dWw+XG5cdHt7ICRkZXRhaWxzIDo9IC5kZXRhaWxzIH19XG5cdHt7IHJhbmdlICRpbmRleCwgJHBldG5hbWUgOj0gLnBldHMgfX1cbiAge3sgJHBldCA6PSBpbmRleCAkZGV0YWlscyAkcGV0bmFtZSB9fVxuICB7eyAkZnVsbG5hbWUgOj0gaW5kZXggJHBldCBcImZ1bGxuYW1lXCIgfX1cbiAgPGxpPlxuICBcdHt7IGlmIGVxICRmdWxsbmFtZSAkcGV0bmFtZSB9fVxuICAgIFx0e3sgJGZ1bGxuYW1lIH19XG4gICAge3sgZWxzZSB9fVxuICAgIFx0e3sgJGZ1bGxuYW1lIH19IChha2Ege3skcGV0bmFtZX19KVxuICAgIHt7IGVuZCB9fVxuICA8L2xpPlxuICB7e2VuZH19XG48dWw+IiwiaW5wdXQiOiJ7XG5cdFwicGV0c1wiOiBbXCJUaXhpZVwiLCBcIkNsZW9cIiwgXCJOZXV0cmlub1wiLCBcIlZlY3RvclwiLCBcIkxlb1wiLCBcIkFyY2hpZVwiXSxcbiAgXCJkZXRhaWxzXCI6IHtcbiAgICBcIlRpeGllXCI6IHtcbiAgICBcdFwiZnVsbG5hbWVcIjogXCJUaXhpZVwiLFxuICAgICAgXCJzcGVjaWVzXCI6IFwiQmVhcmRlZCBEcmFnb25cIixcbiAgICAgIFwibmlja25hbWVzXCI6IFtcIlRyaXhcIl1cbiAgICB9LFxuXHRcdFwiQ2xlb1wiOiB7XG4gICAgXHRcImZ1bGxuYW1lXCI6IFwiQ2xlb2NhdHJhXCIsXG4gICAgICBcInNwZWNpZXNcIjogXCJDYXRcIixcbiAgICAgIFwibmlja25hbWVzXCI6IFtcIkNsZW9cIl1cbiAgICB9LFxuICAgIFwiTmV1dHJpbm9cIjoge1xuICAgIFx0XCJmdWxsbmFtZVwiOiBcIk5ldXRyaW5vXCIsXG4gICAgICBcInNwZWNpZXNcIjogXCJDYXRcIixcbiAgICAgIFwibmlja25hbWVzXCI6IFtcIlRpZ2Vyd29sZlwiXVxuICAgIH0sXG4gICAgXCJWZWN0b3JcIjoge1xuICAgIFx0XCJmdWxsbmFtZVwiOiBcIlZlY3RvclwiLFxuICAgICAgXCJzcGVjaWVzXCI6IFwiQ2F0XCIsXG4gICAgICBcIm5pY2tuYW1lc1wiOiBbXVxuICAgIH0sXG4gICAgXCJMZW9cIjoge1xuICAgIFx0XCJmdWxsbmFtZVwiOiBcIkxlb25oYXJkXCIsXG4gICAgICBcInNwZWNpZXNcIjogXCJEb2dcIixcbiAgICAgIFwibmlja25hbWVzXCI6IFtcIkxlb1wiLCBcIldpZ2dsZXMgTWNHZWVcIiwgXCJNci4gRmxvcHB5IEVhcnNcIiwgXCJTaXIgU25pZmZzLWEtbG90XCJdXG4gICAgfSxcbiAgICBcIkFyY2hpZVwiOiB7XG4gICAgXHRcImZ1bGxuYW1lXCI6IFwiQXJjaGltZWRlc1wiLFxuICAgICAgXCJzcGVjaWVzXCI6IFwiRG9nXCIsXG4gICAgICBcIm5pY2tuYW1lc1wiOiBbXCJBcmNoaWVcIiwgXCJDaGFvc1wiLCBcIk1yIFNxdWlzaHkgRmFjZVwiXVxuICAgIH1cbiAgfVxufSIsImNvbmZpZyI6eyJ0ZW1wbGF0ZSI6InRleHQiLCJmdWxsU2NyZWVuSFRNTCI6ZmFsc2UsImZ1bmN0aW9ucyI6WyJzcHJpZyJdLCJvcHRpb25zIjpbImxpdmUiXX19) ... can lead to incorrect whitespace, like our comma issue
2. [Only `{{- -}}` is **very** bad](https://repeatit.io/#/share/eyJ0ZW1wbGF0ZSI6Int7LypcbkNocmlzIENodWR6aWNraSwgTm92IDE1LCAyMDIyXG5cbk9ubHkge3stIC19fSA9IFdyb25nIGFuZCBiYWRseSBmb3JtYXR0ZWQgb3V0cHV0XG4qL319PHVsPlxuXHR7ey0gJGRldGFpbHMgOj0gLmRldGFpbHMgLX19XG5cdHt7LSByYW5nZSAkaW5kZXgsICRwZXRuYW1lIDo9IC5wZXRzIC19fVxuICB7ey0gJHBldCA6PSBpbmRleCAkZGV0YWlscyAkcGV0bmFtZSAtfX1cbiAge3stICRmdWxsbmFtZSA6PSBpbmRleCAkcGV0IFwiZnVsbG5hbWVcIiAtfX1cbiAgPGxpPlxuICBcdHt7LSBpZiBlcSAkZnVsbG5hbWUgJHBldG5hbWUgLX19XG4gICAgXHR7ey0gJGZ1bGxuYW1lIC19fVxuICAgIHt7LSBlbHNlIC19fVxuICAgIFx0e3stICRmdWxsbmFtZSAtfX0gKGFrYSB7ey0gJHBldG5hbWUgLX19KVxuICAgIHt7LSBlbmQgLX19XG4gIDwvbGk+XG4gIHt7LSBlbmQgLX19XG48dWw+IiwiaW5wdXQiOiJ7XG5cdFwicGV0c1wiOiBbXCJUaXhpZVwiLCBcIkNsZW9cIiwgXCJOZXV0cmlub1wiLCBcIlZlY3RvclwiLCBcIkxlb1wiLCBcIkFyY2hpZVwiXSxcbiAgXCJkZXRhaWxzXCI6IHtcbiAgICBcIlRpeGllXCI6IHtcbiAgICBcdFwiZnVsbG5hbWVcIjogXCJUaXhpZVwiLFxuICAgICAgXCJzcGVjaWVzXCI6IFwiQmVhcmRlZCBEcmFnb25cIixcbiAgICAgIFwibmlja25hbWVzXCI6IFtcIlRyaXhcIl1cbiAgICB9LFxuXHRcdFwiQ2xlb1wiOiB7XG4gICAgXHRcImZ1bGxuYW1lXCI6IFwiQ2xlb2NhdHJhXCIsXG4gICAgICBcInNwZWNpZXNcIjogXCJDYXRcIixcbiAgICAgIFwibmlja25hbWVzXCI6IFtcIkNsZW9cIl1cbiAgICB9LFxuICAgIFwiTmV1dHJpbm9cIjoge1xuICAgIFx0XCJmdWxsbmFtZVwiOiBcIk5ldXRyaW5vXCIsXG4gICAgICBcInNwZWNpZXNcIjogXCJDYXRcIixcbiAgICAgIFwibmlja25hbWVzXCI6IFtcIlRpZ2Vyd29sZlwiXVxuICAgIH0sXG4gICAgXCJWZWN0b3JcIjoge1xuICAgIFx0XCJmdWxsbmFtZVwiOiBcIlZlY3RvclwiLFxuICAgICAgXCJzcGVjaWVzXCI6IFwiQ2F0XCIsXG4gICAgICBcIm5pY2tuYW1lc1wiOiBbXVxuICAgIH0sXG4gICAgXCJMZW9cIjoge1xuICAgIFx0XCJmdWxsbmFtZVwiOiBcIkxlb25oYXJkXCIsXG4gICAgICBcInNwZWNpZXNcIjogXCJEb2dcIixcbiAgICAgIFwibmlja25hbWVzXCI6IFtcIkxlb1wiLCBcIldpZ2dsZXMgTWNHZWVcIiwgXCJNci4gRmxvcHB5IEVhcnNcIiwgXCJTaXIgU25pZmZzLWEtbG90XCJdXG4gICAgfSxcbiAgICBcIkFyY2hpZVwiOiB7XG4gICAgXHRcImZ1bGxuYW1lXCI6IFwiQXJjaGltZWRlc1wiLFxuICAgICAgXCJzcGVjaWVzXCI6IFwiRG9nXCIsXG4gICAgICBcIm5pY2tuYW1lc1wiOiBbXCJBcmNoaWVcIiwgXCJDaGFvc1wiLCBcIk1yIFNxdWlzaHkgRmFjZVwiXVxuICAgIH1cbiAgfVxufSIsImNvbmZpZyI6eyJ0ZW1wbGF0ZSI6InRleHQiLCJmdWxsU2NyZWVuSFRNTCI6ZmFsc2UsImZ1bmN0aW9ucyI6WyJzcHJpZyJdLCJvcHRpb25zIjpbImxpdmUiXX19) Can lead to  wrong results (e.g., no space between classnames—very bad—or no space around dash). Also leads to scrunched output.
3. [Good: Thoughtful use of `{{}}` and `{{- -}}`](https://repeatit.io/#/share/eyJ0ZW1wbGF0ZSI6Int7LypcbkNocmlzIENodWR6aWNraSwgTm92IDE1LCAyMDIyXG5cblVzZSBib3RoIHt7IH19IGFuZCB7ey0gLX19IHRob3VnaHRmdWxseSA9IEdvb2Qgb3V0cHV0XG4qL319PHVsPlxuXHR7ey0gJGRldGFpbHMgOj0gLmRldGFpbHMgLX19XG5cdHt7LSByYW5nZSAkaW5kZXgsICRwZXRuYW1lIDo9IC5wZXRzIC19fVxuICB7ey0gJHBldCA6PSBpbmRleCAkZGV0YWlscyAkcGV0bmFtZSAtfX1cbiAge3svKlxuICBcdEFMRVJUOiBPbiB0aGUgbmV4dCBsaW5lLCBubyByaWdodC1oYW5kIHRyaW0uXG4gICAgVGhpcyBrZWVwcyA8dWw+IGFuZCA8bGk+IG9uIHNlcGFyYXRlIGxpbmVzIHdpdGggbmljZSBpbmRlbnRhdGlvbi5cbiAgICBQcmlvcml0eTogTG93XG4gICAgXG4gICAgVGhpcyBjb21tZW50IGRvZXMgbm90IG5lZWQgdHJpbSAoYnV0IHdvdWxkIG5vdCBodXJ0KSBzaW5jZSBib3RoIHRhZ3MgdG8gbGVmdCBhbmQgcmlnaHRcbiAgICBhcmUgdHJpbW1lZC5cbiAgKi99fVxuICB7ey0gJGZ1bGxuYW1lIDo9IGluZGV4ICRwZXQgXCJmdWxsbmFtZVwiIH19XG4gIDxsaT5cbiAgXHR7eyBpZiBlcSAkZnVsbG5hbWUgJHBldG5hbWUgLX19XG4gICAgXHR7eyAkZnVsbG5hbWUgfX1cbiAgICB7ey0gZWxzZSAtfX1cbiAgICBcdHt7LSAvKlxuICAgICAgQUxFUlQ6IENhbm5vdCB0cmltIHJpZ2h0IHNpZGUgJGZ1bGxuYW1lIGJlbG93LCBvciB3ZSBsb3NlIG1lYW5pbmdmdWwgd2hpdGVzcGFjZS5cbiAgICAgIChUcmltbWluZyBsZWZ0IHNpZGUgaXMgT0sgYnV0IGRvZXMgbm90aGluZy4pXG4gICAgICBQcmlvcml0eTogSGlnaFxuICAgICAgKi8gLX19XG4gICAgXHR7eyAkZnVsbG5hbWUgfX0gKGFrYSB7eyRwZXRuYW1lfX0pXG4gICAge3stIGVuZCB9fVxuICA8L2xpPlxuICB7ey0gZW5kIH19XG48dWw+IiwiaW5wdXQiOiJ7XG5cdFwicGV0c1wiOiBbXCJUaXhpZVwiLCBcIkNsZW9cIiwgXCJOZXV0cmlub1wiLCBcIlZlY3RvclwiLCBcIkxlb1wiLCBcIkFyY2hpZVwiXSxcbiAgXCJkZXRhaWxzXCI6IHtcbiAgICBcIlRpeGllXCI6IHtcbiAgICBcdFwiZnVsbG5hbWVcIjogXCJUaXhpZVwiLFxuICAgICAgXCJzcGVjaWVzXCI6IFwiQmVhcmRlZCBEcmFnb25cIixcbiAgICAgIFwibmlja25hbWVzXCI6IFtcIlRyaXhcIl1cbiAgICB9LFxuXHRcdFwiQ2xlb1wiOiB7XG4gICAgXHRcImZ1bGxuYW1lXCI6IFwiQ2xlb2NhdHJhXCIsXG4gICAgICBcInNwZWNpZXNcIjogXCJDYXRcIixcbiAgICAgIFwibmlja25hbWVzXCI6IFtcIkNsZW9cIl1cbiAgICB9LFxuICAgIFwiTmV1dHJpbm9cIjoge1xuICAgIFx0XCJmdWxsbmFtZVwiOiBcIk5ldXRyaW5vXCIsXG4gICAgICBcInNwZWNpZXNcIjogXCJDYXRcIixcbiAgICAgIFwibmlja25hbWVzXCI6IFtcIlRpZ2Vyd29sZlwiXVxuICAgIH0sXG4gICAgXCJWZWN0b3JcIjoge1xuICAgIFx0XCJmdWxsbmFtZVwiOiBcIlZlY3RvclwiLFxuICAgICAgXCJzcGVjaWVzXCI6IFwiQ2F0XCIsXG4gICAgICBcIm5pY2tuYW1lc1wiOiBbXVxuICAgIH0sXG4gICAgXCJMZW9cIjoge1xuICAgIFx0XCJmdWxsbmFtZVwiOiBcIkxlb25oYXJkXCIsXG4gICAgICBcInNwZWNpZXNcIjogXCJEb2dcIixcbiAgICAgIFwibmlja25hbWVzXCI6IFtcIkxlb1wiLCBcIldpZ2dsZXMgTWNHZWVcIiwgXCJNci4gRmxvcHB5IEVhcnNcIiwgXCJTaXIgU25pZmZzLWEtbG90XCJdXG4gICAgfSxcbiAgICBcIkFyY2hpZVwiOiB7XG4gICAgXHRcImZ1bGxuYW1lXCI6IFwiQXJjaGltZWRlc1wiLFxuICAgICAgXCJzcGVjaWVzXCI6IFwiRG9nXCIsXG4gICAgICBcIm5pY2tuYW1lc1wiOiBbXCJBcmNoaWVcIiwgXCJDaGFvc1wiLCBcIk1yIFNxdWlzaHkgRmFjZVwiXVxuICAgIH1cbiAgfVxufSIsImNvbmZpZyI6eyJ0ZW1wbGF0ZSI6InRleHQiLCJmdWxsU2NyZWVuSFRNTCI6ZmFsc2UsImZ1bmN0aW9ucyI6WyJzcHJpZyJdLCJvcHRpb25zIjpbImxpdmUiXX19) Correct output, good formatting, no extra whitespace.
